core/wiki/macros/keyboard-driven-input.tid

```diff
@@ -0,0 +1,40 @@
+title: $:/core/macros/keyboard-driven-input
+tags: $:/tags/Macro
+
+\define input-next-actions(afterbefore:"after",reverse:"",tiddler,storeTitle,field,index,filter1,filter2,refreshTitle)
+<$list filter="[<__storeTitle__>get[text]minlength<__filterMinLength__>] [<__filterMinLength__>match[0]] +[limit[1]]" variable="ignore">
+<$vars userInput={{{ [<__storeTitle__>get[text]] }}} selectedItem={{{ [<selectionTiddler>get[text]] }}}>
+<$set name="filteredList" filter="[subfilter<filter1>addsuffix[-list1]] =[subfilter<filter2>addsuffix[-list2]]">
+<$set name="nextItem" value={{{ [enlist<filteredList>$afterbefore$<selectedItem>] ~[enlist<filteredList>$reverse$nth[1]] }}}>
+<$list filter="[<nextItem>minlength[1]]">
+<$action-setfield $tiddler=<<selectionTiddler>> text=<<nextItem>>/>
+<$list filter="[<__index__>match[]]">
+<$action-setfield $tiddler=<<__tiddler__>> $field=<<__field__>> $value={{{ [<nextItem>] +[splitregexp[(?:.(?!-))+$]] }}}/>
+<$action-setfield $tiddler=<<__refreshTitle__>> text="yes"/>
+</$list>
+<$list filter="[<__index__>!match[]]">
+<$action-setfield $tiddler=<<__tiddler__>> $index=<<__index__>> $value={{{ [<nextItem>] +[splitregexp[(?:.(?!-))+$]] }}}/>
+<$action-setfield $tiddler=<<__refreshTitle__>> text="yes"/>
+</$list>
+</$list>
+</$set>
+</$set>
+</$vars>
+</$list>
+\end
+
+\define keyboard-driven-input(tiddler,storeTitle,field:"text",index:"",tag:"input",type,focus:"",inputactions,input-accept-actions,input-accept-variant-actions,input-cancel-actions,placeholder:"",default:"",class,filter1,filter2,focusPopup,rows,minHeight,tabindex,size,autoHeight,filterMinLength:"0",refreshTitle,selectionTiddler)
+<$set name="selectionTiddler" value="$selectionTiddler$">
```

Comment:
Can this set widget be avoided by directly using the selectionTiddler parameter with `<__selectionTiddler__>`
This is an artifact from when I was testing removing the indexed operator while wanting to make minimal changes.

Reply:
yeah, I was looking at this before... I'll have another, deeper look :grin:

Reply:
PS: If you can post an updated demo at the end of the day when you are done tweaking, that would be helpful. I am realizing that collaborating on a pull request on github and keeping files up to date is tricky.

Reply:
If we avoid the set widget, then it must be passed to the input-nex-actions macro ... the second is more elegant so I'll change it

Reply:
Yes I'll post an up-to-date demo later :)

Reply:
@BurningTreeC many thanks!